src/Connection/ConnectionLimitingPool.php

```diff
@@ -170,6 +176,21 @@ private function getStreamFor(string $uri, Request $request, CancellationToken $
                 $stream = yield $this->getStreamFromConnection($connection, $request);
 
                 if ($stream === null) {
+                    $connectionId = \spl_object_id($connection);
+
+                    \assert(
+                        !isset($this->activeRequestCounts[$connectionId])
+                        || $this->activeRequestCounts[$connectionId] >= 0
+                    );
+
+                    if (!$this->isAdditionalConnectionAllowed($uri)
+                        && ($this->activeRequestCounts[$connectionId] ?? 0) === 0
+                    ) {
+                        // No additional connections allowed, but this connection is idle and unsuited for this request.
```

Comment:
Instead of the comment, we could refactor the second condition of the `if` into an `$this->isIdle($connection)` method.